bedevere/bpo.py

```diff
@@ -40,16 +34,21 @@
     issue_number_found = ISSUE_RE.search(event.data["pull_request"]["title"])
     if not issue_number_found:
         issue = await util.issue_for_PR(gh, event.data["pull_request"])
-        status = SKIP_ISSUE_STATUS if util.skip("issue", issue) else FAILURE_STATUS
+        status = (SKIP_ISSUE_STATUS if util.skip("issue", issue)
+                                    else create_failure_status_no_issue())
     else:
+        issue_number = issue_number_found.group("issue")
         if "body" in event.data["pull_request"]:
             body = event.data["pull_request"]["body"] or ""
             if not body or CLOSING_TAG not in body:
-                issue_number = issue_number_found.group("issue")
                 new_body = BODY.format(body=body, issue_number=issue_number)
                 body_data = {"body": new_body, "maintainer_can_modify": True}
                 await gh.patch(event.data["pull_request"]["url"], data=body_data)
-        status = create_success_status(issue_number_found)
+        issue_number_on_bpo = await _validate_issue_number(issue_number)
+        if issue_number_on_bpo:
+            status = create_success_status(issue_number_found)
```

Comment:
Any reason to not pass in `issue_number` since you already have that pulled out?

Reply:
When I created the two functions `create_failure_status_issue_not_on_bpo` and `create_failure_status_no_issue`, I wanted them to be consistent with how `create_success_status` was already doing things. That's why I passed `issue_number_found` as a parameter and not `issue_number`. I agree that it would be better to pass `issue_number` though, since that is already there.